lib/ecto/changeset/relation.ex

```diff
@@ -39,6 +39,13 @@ defmodule Ecto.Changeset.Relation do
   def empty?(%{}, _), do: false
 
   @doc """
+  Filter empty changes
+  """
+  def filter_empty(%{cardinality: :many}, changes) do
```

Comment:
Can you please make `empty?` use this function for the `cardinality: :many` case and then compare against an empty list?